tensorrt_llm/_torch/auto_deploy/transformations/library/rope.py

```diff
@@ -50,13 +50,126 @@ def apply_rotary_emb(
 from typing import Any, DefaultDict, Dict, List, Optional, Sequence
 
 import torch
+from torch._inductor.pattern_matcher import Match, PatternMatcherPass
```

Comment:
nit: can you `Match` and `PatternMatcherPass` to `utils.pattern_matcher_utils` so we can import it all from the same file. Seems like we will need those imports a lot and I'd rather avoid importing from a private torch file everywhere